register-system.yaml

```diff
@@ -189,3 +199,45 @@
       delegate_to: 127.0.0.1
       loop: "{{ ansible_interfaces }}"
       when: interface.type == "ether" and interface.ipv4.address is defined
+
+    - name: "Pause for some time so that lldptool has enough time to capture necessary information"
+      pause:
+        minutes: 5
+
+    - name: "Get switchport name that's connected to NIC"
+      shell: >-
+        lldptool -i {{ item }} get-tlv -n -V portID | awk -F": " '/Ifname|Local/ {print $2}'
+      failed_when: false
+      loop: "{{ physical_interfaces.stdout_lines }}"
+      register: port_ids
+
+    - name: "Get hostname of the switch that a NIC is connected to"
+      shell: >-
+        lldptool -i {{ item.item }} get-tlv -n -V sysName
+      failed_when: false
+      loop: "{{ port_ids.results }}"
+      register: sysnames
+
+    - set_fact:
+        triplets: "{{ triplets + [[item.item.item, item.item.stdout, item.stdout.split('\t')[1] ]] }}"
+      when: item.rc == 0
+      loop: "{{ sysnames.results }}"
+      vars:
+        triplets: []
+
+    - name: "Create cables that connect NIC to switchports"
+      netbox.netbox.netbox_cable:
+        netbox_url: "{{ netbox_url }}"
+        netbox_token: "{{ netbox_token }}"
+        data:
+          termination_a_type: dcim.interface
+          termination_a:
+            device: "{{ inventory_hostname }}"
+            name: "{{ item[0] }}"
+          termination_b_type: dcim.interface
+          termination_b:
+            device: "{{ item[2] }}"
+            # Cisco switchports are registersted like "Ethernet X/Y" but lldp data has it as Eth X/Y
+            name: "{{ item[1].replace('Eth', 'Ethernet') if 'Ethernet' not in item[1] else item[1]}}"
```

Comment:
I feel like we should probably perform this transformation in the `set_fact` task.